### PR TITLE
Fix duckdb array type

### PR DESCRIPTION
**Pull request by eakmanrq**
https://duckdb.org/docs/sql/data_types/overview

DuckDB expects that array are declared using `INTERIOR[]` instead of `ARRAY<INTERIOR>`.

### CODE REVIEW

File: sqlglot/dialects/duckdb.py [hunk 1]
```diff
@@ -68,6 +68,15 @@ def _struct_pack_sql(self, expression):
     return f"STRUCT_PACK({', '.join(args)})"
 
 
+def _datatype_sql(self, expression):
+    type_value = expression.this
+    if type_value == exp.DataType.Type.ARRAY:
+        interior = self.expressions(expression, flat=True)
```
Comment (tobymao) [inline sqlglot/dialects/duckdb.py:74]: sorry to nit but i'm not a fan of this style can we just do

if expression.this == exp.DataType.Type.ARRAY:
    return f"{self.expressions(expression, flat=True)}[]"
return self.datatype_sql(expression)

File: sqlglot/parser.py [hunk 2]
```diff
@@ -1676,6 +1676,11 @@ def _parse_types(self):
         is_struct = type_token == TokenType.STRUCT
         expressions = None
 
+        if not nested and self._match_pair(TokenType.L_BRACKET, TokenType.R_BRACKET):
```
Comment (tobymao) [inline sqlglot/parser.py:1679]: what is this?

Comment (tobymao) [reply sqlglot/parser.py:1679]: is this if you do int[] which means int array?

Comment (eakmanrq) [reply sqlglot/parser.py:1679]: Yeah `int[]` means an int array.

File: sqlglot/parser.py [hunk 3]
```diff
@@ -1676,6 +1676,11 @@ def _parse_types(self):
...
+            return exp.DataType(
+                this=exp.DataType.Type.ARRAY, expressions=[exp.DataType.build(type_token.value.upper())], nested=True
```
Comment (tobymao) [inline sqlglot/parser.py:1681]: you don't need to call upper here

**Pull request merged**